lxd-qemu-snap/snapcraft.yaml

```diff
@@ -0,0 +1,235 @@
+name: qemu-for-lxd
+base: core24
+assumes:
+  - snapd2.63
+version: '1'
+grade: devel
+summary: External Qemu for LXD
+description: |-
+ External Qemu for LXD
+
+contact: lxd@lists.canonical.com
+issues: https://github.com/canonical/lxd/issues
+source-code: https://github.com/canonical/lxd
+website: https://ubuntu.com/lxd
+confinement: strict
+
+slots:
+  qemu-external:
+    interface: content
+    content: qemu-external-binaries
+    read:
+      - $SNAP/external/qemu
+
+parts:
+  #
+  # This step is extremely important, as it ensures that no mesa libs will
+  # be provided with this snap to prevent any possible version mistmatches
+  # between this snap and mesa-2404 snap.
+  # If you the error like:
+  # DRI driver not from this Mesa build ('24.0.5-1ubuntu1' vs '24.0.9-0ubuntu0.1')
+  # this is a clear sign that you likely ship some mesa library along with your qemu-for-lxd
+  # snap and it has a version conflict with what you have in gpu-2404 interface snap.
+  #
+  gpu-2404:
+    after:
+      - virgl
+      - qemu
+    source: https://github.com/canonical/gpu-snap.git
+    plugin: dump
+    override-prime: |
+      set -ex
+
+      CUSTOM_PREFIX="external/qemu"
+      CRAFT_PRIME_OVERRIDE="${CRAFT_PRIME}/${CUSTOM_PREFIX}"
+
+      # cleanup script expects to see libraries in usr/lib, while we have them in lib
+      mkdir "${CRAFT_PRIME_OVERRIDE}/usr"
+      ln -s "${CRAFT_PRIME_OVERRIDE}/lib" "${CRAFT_PRIME_OVERRIDE}/usr/lib"
```

Comment:
Would doing this via the [`organize:`](https://snapcraft.io/docs/snapcraft-parts-metadata#organize) section work?

Reply:
Can we create a symlink using the `organize` step? I haven't know about that.

Reply:
No, it seems to only do renames.

Reply:
ah, then I think the `organize` thing is not very helpful in this case. We don't want to have `usr/lib` at the end. It's just a temporary hack to provide a cleanup script with what it expects to see. At the end we remove `usr/lib` symlink and all libraries live in `lib`.